### DISCUSSION TITLE
Is it possible to set X11's "Extended Window Manager Hints" when using the eframe?

### DISCUSSION THREAD

**Discussion by x7Gv** [emilk/egui#1432] [Q&A] [answered]
Hello.

I would like to use egui through eframe for making an overlay-like application.
I managed to find out that this kind of app would require transparent background without window decorations and so on, which seems to be supported via `eframe::NativeOptions` options. When I tested the app on my personal system, which uses a tiling window manager on top of X11, I couldn't find a solution to the problem of the window being spawned as a tiled one by default (that is the WM isn't instructed by the end user to spawn it floating).

I found out that this state in X11 protocols case could be altered by setting [these flags](https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm140200472629520).
This seems to be possible in winit, which I don't know whether or not the eframe uses, but in that case I couldn't deduce how to actually set it in an eframe scenario.

So the main question is as it is in the title: 
Is it possible to set X11's "Extended Window Manager Hints" when using the eframe?

Thank you for your time.

**Comment by coderedart** [comment, marked as answer]
I may sound a little negative, but its just my frustration with the overlay work :D 
1. what you are talking about is always_on_top which is available in Native options. 
2. that is really really not enough to make an overlay. the most important thing is to allow mouse clicks to passthrough while your application window is still on top.
3. https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#STACKINGORDER which you are talking about (i think) has the following words `	the following layered stacking order is recommended`. so, its not guaranteed. for example, in i3, floating windows are always on top on tiling windows, but you cannot set one floating window to be above another.https://github.com/i3/i3/issues/4478
4. finally, winit doesn't yet allow you to enable passthrough feature. https://github.com/rust-windowing/winit/pull/2232 

you will need to make your own egui integration. I made a thing for this https://github.com/coderedart/egui_overlay which uses glfw + wgpu + egui to make an overlay. you can fork it and just edit the code to play around. there's already an example in that repo too which you can run with 

`cargo run --release --example=hello`

its also crossplatform (atleast windows + linux). don't bother with x11, I have spent months before i could make it work (and a lot of x11 spec reading). use the egui_overlay for now. if hit-test feature of winit i linked about is finally allowed, then i will probably port the egui_overlay to winit. 

also, fullscreen windows are often below always_on_top windows. you need to set a `TRANSIENT_FOR` hint on your window with the window id of the fullscreen application you would like to overlay upon and other such tiny details which i'm still working out. 

I'm making an overlay for Guild Wars 2 game, and its been months of work to find all the little quirks.

**Reply by coderedart** [reply]
you can find a lot of my journey documented here https://gitlab.com/coderedart/jokolay/-/issues/1
